Change '*' to '...' and `...` for named tensor API functions.

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26350 Change '*' to '...' and `...` for named tensor API functions.**
* #26349 Rename torch.namedtensor -> torch._namedtensor_internals

Python 3 lets us use `...` to perform indexing. Semantically, `...`
means "the rest of the unspecified dimensions". For example, while
indexing, one can do (for 5D `tensor`) `tensor[0, 0, ..., 0]` and
the `...` is expanded into `tensor[0, 0, :, :, 0]`.

Previously, we were using '*' to represent a similar behavior in names.
For example, `tensor.refine_names` supports things like the following:

```
x = torch.randn(2, 3, 4, 5, 6)
x_out = x.refine_names('*', 'H', 'W')  # refine only the last two
dimensions
```

This PR changes it so that named tensor API functions recognize `'...'`
(in Python 2 and Python 3) and `...` (in Python 3 exclusively) instead
of `'*'`.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17424666](https://our.internmc.facebook.com/intern/diff/D17424666)

zou3519 committed:
Change '*' to '...' and `...` for named tensor API functions.

Python 3 lets us use `...` to perform indexing. Semantically, `...`
means "the rest of the unspecified dimensions". For example, while
indexing, one can do (for 5D `tensor`) `tensor[0, 0, ..., 0]` and
the `...` is expanded into `tensor[0, 0, :, :, 0]`.

Previously, we were using '*' to represent a similar behavior in names.
For example, `tensor.refine_names` supports things like the following:

```
x = torch.randn(2, 3, 4, 5, 6)
x_out = x.refine_names('*', 'H', 'W')  # refine only the last two
dimensions
```

This PR changes it so that named tensor API functions recognize `'...'`
(in Python 2 and Python 3) and `...` (in Python 3 exclusively) instead
of `'*'`.

Test Plan:
- [namedtensor ci]

nairbv commented on test/test_namedtensor.py:
```diff
@@ -302,35 +309,35 @@ def test_renamed_globber(self):
-        self.assertEqual(named_tensor.renamed('*').names, named_tensor.names)
-        self.assertEqual(named_tensor.renamed('*', 'width').names,
+        self.assertEqual(named_tensor.renamed('...').names, named_tensor.names)
+        self.assertEqual(named_tensor.renamed('...', 'width').names,
```
in python3 these all work without the quotes?

zou3519 replied:
Yes! There's an explicit python-3 test for this behavior somewhere in this file

nairbv reviewed (approved):
Cool!

zou3519 committed:
Update on "Change '*' to '...' and `...` for named tensor API functions."


Python 3 lets us use `...` to perform indexing. Semantically, `...`
means "the rest of the unspecified dimensions". For example, while
indexing, one can do (for 5D `tensor`) `tensor[0, 0, ..., 0]` and
the `...` is expanded into `tensor[0, 0, :, :, 0]`.

Previously, we were using '*' to represent a similar behavior in names.
For example, `tensor.refine_names` supports things like the following:

```
x = torch.randn(2, 3, 4, 5, 6)
x_out = x.refine_names('*', 'H', 'W')  # refine only the last two
dimensions
```

This PR changes it so that named tensor API functions recognize `'...'`
(in Python 2 and Python 3) and `...` (in Python 3 exclusively) instead
of `'*'`.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17424666](https://our.internmc.facebook.com/intern/diff/D17424666)

zou3519 committed:
Update on "Change '*' to '...' and `...` for named tensor API functions."


Python 3 lets us use `...` to perform indexing. Semantically, `...`
means "the rest of the unspecified dimensions". For example, while
indexing, one can do (for 5D `tensor`) `tensor[0, 0, ..., 0]` and
the `...` is expanded into `tensor[0, 0, :, :, 0]`.

Previously, we were using '*' to represent a similar behavior in names.
For example, `tensor.refine_names` supports things like the following:

```
x = torch.randn(2, 3, 4, 5, 6)
x_out = x.refine_names('*', 'H', 'W')  # refine only the last two
dimensions
```

This PR changes it so that named tensor API functions recognize `'...'`
(in Python 2 and Python 3) and `...` (in Python 3 exclusively) instead
of `'*'`.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17424666](https://our.internmc.facebook.com/intern/diff/D17424666)
